.gitlab-ci.yml

```diff
@@ -1,14 +1,33 @@
 variables:
-  IMAGE_REPOSITORY: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/test-infra-definitions
+  BUILD_STABLE_REGISTRY: 486234852809.dkr.ecr.us-east-1.amazonaws.com
+  CI_IMAGE_REPO: "ci/${CI_PROJECT_NAME}"
+  CI_REGISTRY_IMAGE: "$BUILD_STABLE_REGISTRY/$CI_IMAGE_REPO/runner"
+  CI_REGISTRY_IMAGE_TEST: "$BUILD_STABLE_REGISTRY/$CI_IMAGE_REPO/runner-dev"
   KUBERNETES_SERVICE_ACCOUNT_OVERWRITE: "test-infra-definitions"
 
 stages:
-  - test
   - build
+  - test
+  - release
+
+
+build-runner-image:
+  stage: build
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/docker:20.10.13
+  tags: ["arch:amd64"]
+  variables:
+    RELEASE_IMAGE: "false"
+  rules:
+    - when: on_success
+  script:
+    # Tag with 8 and 12 SHA to match go.mod
+    - docker buildx build --no-cache --pull --push --label target=build --tag ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA} .
```

Comment:
I'm not sure the `label target=build` brings any value, image metadata are almost never used.

Reply:
It is mentioned in the CI/CD onboarding to Gitlab runners: https://datadoghq.atlassian.net/wiki/spaces/DEVX/pages/2411528655/Onboarding+To+Gitlab+CI#4.-ECR-(Elastic-Container-Registry)-for-a-CI-Image-(optional)

Reply:
It's dedicated to `registry.dd.io` not ECR.